src/clients/c++/library/request_http.cc

```diff
@@ -709,6 +731,13 @@ SharedMemoryControlHttpContextImpl::SendRequest(
   if (action_str == "register") {
     full_url += +"/" + name + "/" + shm_key + "/" + std::to_string(offset) +
                 "/" + std::to_string(byte_size);
+  } else if (action_str == "cudaregister") {
+#if TRTIS_ENABLE_GPU
+    // TODO set cuda-shm_handle as raw bytes in body
```

Comment:
Still have a TODO. Also shm_key is not used for cuda shared. Using the common SendRequest with the union of all the necessary arguments is confusing. The non-common part is the URL creation. So move that into the calling functions and have SendRequest just take the url.

Reply:
This is fixed in the client side changes for cuda shared memory, that is a different PR here: https://github.com/NVIDIA/tensorrt-inference-server/pull/756